inginious/frontend/plugins/notebooks_grader/api/notebook_grader_api.py

```diff
@@ -224,9 +241,14 @@ def API_POST(self): # pylint: disable=arguments-differ
                 test = results[test_id]
                 self.__validate_signature(username, test["id"], test["signature"])
             #We calculate the submissions grade based on the test grades, 
-            #for now every test has the same weight
-            grades = [results[test_id]["test_grade"] for test_id in results]
-            submission_grade = sum(grades) / len(grades)
+            #we calculate weighted average
+            tests_weights = self.__get_tests_weights(course_id,task_id)
+            #grades is a list of tuples containing (grade, weight)
+            grades = [(results[test_id]["test_grade"], tests_weights[test_id]) for test_id in results]
+            
+            submission_grade = sum([grade[0]*grade[1] for grade in grades]) / sum([tests_weights[test_id] for test_id in results])
```

Comment:
Please include a validation in case all weigths are assigned to 0 by the teacher. This may raise a ZeroDivisionError exception.